### DISCUSSION TITLE
When does Weblate add news languages to a component?

### DISCUSSION THREAD

**Discussion by luebbe** [WeblateOrg/weblate#5811] [Q&A] [answered]
Given the following configuration:
- User interface with six languages
- User manual with three languages (subset of the six UI languages)

The manual will never(*) be translated into the three other languages.

After setting up the project, I see six languages here and three languages there with a translation rate of say 95% each.
At some point Weblate decides to add the "missing" three languages to the manual and the manual's translation rate drops to <50%. 

- When and why does this happen? I didn't knowingly trigger anything. Does activating an extension trigger this?
- How can I prevent this from happening?

(*) never say never ;)

**Comment by nijel** [comment, marked as answer]
Language consistency add-on will do this, or users can add language when this is enabled.

**Reply by luebbe** [reply]
OK, so this means that I shouldn't activate the language consistency add-on. That's a pity.
Wouldn't it make sense for this add-on to have an operation mode where it "just" checks for consistency across existing translations instead of adding empty translations on its own? 
Especially when adding new translations is disabled for users. Why should an add on be allowed to override this?

**Reply by nijel** [reply]
The add-on only adds new languages, it does nothing else

**Reply by luebbe** [reply]
Yes, but these undesired empty languages distort the statistics. They are also in the downloaded .zip file and are probably pushed to the repository too.
This could for example lead to a build system spitting out a French manual with all English text. The build system doesn't necessarily know anything about the quality of the available translations. It just collects all .po files and starts building.

**Reply by nijel** [reply]
Okay, I was probably not clear enough:

> Wouldn't it make sense for this add-on to have an operation mode where it "just" checks for consistency across existing translations instead of adding empty translations on its own?

The addon does not check whether things are translated consistently, it only ensures that a consistent set of languages is used in all components. 

See https://docs.weblate.org/en/latest/admin/addons.html#language-consistency and https://docs.weblate.org/en/latest/admin/continuous.html#keeping-translations-same-across-components

**Reply by luebbe** [reply]
Thank you. I read the docs, more than once, but I really understood it wrong.

**Reply by nijel** [reply]
The addon name should be probably better, let's deal with this in https://github.com/WeblateOrg/weblate/issues/5829
